### PR TITLE
Update micropython to 1.18

### PR DESCRIPTION
**Pull request by Tunas1337**
I took the liberty of upgrading the microPython version to 1.18.
According to the changelog: 
>     This release of MicroPython sees a boost to the overall performance of the
>     VM and runtime.  This is achieved by the addition of an optional cache to
>     speed up general hash table lookups, as well as a fast path in the VM for
>     the LOAD_ATTR opcode on instance types.  The new configuration options are
>     MICROPY_OPT_MAP_LOOKUP_CACHE and MICROPY_OPT_LOAD_ATTR_FAST_PATH.  As part
>     of this improvement the MICROPY_OPT_CACHE_MAP_LOOKUP_IN_BYTECODE option has
>     been removed, which provided a similar map caching mechanism but with the
>     cache stored in the bytecode, which made it not useful on bare metal ports.
>     The new mechanism is measured to be at least as good as the old one,
>     applies to more map lookups, has a constant RAM overhead, and applies to
>     native code as well as bytecode.
I'm unsure if these are applied by default, but I'm open to someone reviewing this commit to enable any optimizations that would be nice to have.

Yes, it does compile and function. See below:

### CODE REVIEW

**Comment by Lauryy06** [comment]
Does the matplotlib module still work? There were some problems when we upgraded to version 1.17.

File: python/port/genhdr/qstrdefs.in.h [hunk 1]
```diff
@@ -266,22 +263,20 @@ Q(lstrip)
 Q(map)
 Q(math)
 Q(matplotlib)
-Q(matplotlib.pyplot)
+Q(matplotlib_dot_pyplot)
```
Comment (Yaya-Cout) [inline python/port/genhdr/qstrdefs.in.h:266]:
```suggestion
Q(matplotlib.pyplot)
```

Comment (Tunas1337) [reply python/port/genhdr/qstrdefs.in.h:266]: I'll try to compile this and see if it works. Yesterday I tested matplotlib and it didn't exist as a module, so I hope this fixes it.

Comment (Yaya-Cout) [reply python/port/genhdr/qstrdefs.in.h:266]: I already try to upgrade it, but I have an attribute error… #135

Comment (Tunas1337) [reply python/port/genhdr/qstrdefs.in.h:266]: I didn't see your PR, my bad. I suppose I'll look into fixing that, if I can. It's either that, or we wait for Epsilon to actually update to 1.18 and see how they do it.

**Comment by Lauryy06** [comment]
It's done in #259.

**Pull request closed without merging**